### PR TITLE
Use shared Spinner, FullScreenSpinner throughout

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR replaces remaining uses of local LMS `Spinner` and `FullScreenSpinner` with shared components from `frontend-shared`.

~~In draft until SVG types from shared package can be resolved (CI failure).~~ Fixed.

## After

I have my browser throttled to "Fast 3G" for the GIFs here; typically users would not see these spinner for this long.

In the content item selection flow, while launching a picker, and then again after choosing an item:

lms-full-screen-spinner

In the grading flow: a smaller spinner inline in the grading input field itself while a grade is being fetched (this is an unusual UI pattern but I've retained what was there) and a full-screen spinner when a grade is being set:

lms-grading-spinners

## Notes about Implementation and Tests

This one's a head-scratcher!

This is a branch in which I am attempting to remove the last of the local `Spinner` and `FullScreenSpinner` stuff from LMS in favor of shared variants.

I was _this_ close to success when I ran into an oddity that I need a second set of eyes on.

Dropping in the shared `FullScreenSpinner` and removing the local implementation was basically effortless. The `SubmitGradeForm` component gave me a little heartburn, especially with replacing a local `Spinner` usage. My first grief was that I had to refactor a bunch of tests. But then I was triumphant!

But then something weird happened when I tried to remove unused modules. Removing local `FullScreenSpinner` and its test module went just fine. But when I tried to remove `Spinner-test` (test module for the now-unused local `Spinner` implementation), tests for `SubmitGradeForm` related to the (shared) `Spinner` started failing.

~~I've spent about 45 minutes poking at what might be causing this, wandering down a few theoretical paths, but nope. I fear I might be overlooking something obvious. Can someone help me out by checking out this branch, running tests (seeing them pass, I should hope), then removing `Spinner-test.js` and seeing them fail and helping me spot what might be seriously "well, duh?"~~

_Edited_: I've been able to get tests to pass. My hunch is that the switch to the non-mocked shared `Spinner` had some sort of impact on render and component updating. The very act of _importing_ the local `Spinner` component (or not) in the old `Spinner-test.js` module was the catalyst for this behavior change. You can see the change I had to make [in this commit](https://github.com/hypothesis/lms/pull/3374/commits/dfde12e39d075f3e9a1a1b41b3548b96b42cc432) to make tests pass without the presence of  the local `Spinner` import.

### CODE REVIEW

**Comment by robertknight** [comment]
Interesting! - It looks like the issue relates to the scope hoisting Rollup does. This is where it combines the code from the top level of each module into a single module and then renames any symbols whose names would conflict as a result. Having two components of the same name in the bundle forces one of them to be renamed.

Logging the output of `wrapper.debug()` at the top of the `waitFor` callback that you modified shows the difference.

Without local `Spinner` component + tests in the bundle (tests fail):

```
        <span className="SubmitGradeForm__grade-wrapper">
          <input className="SubmitGradeForm__grade" disabled={false} id="SubmitGradeForm__grade:1" onInput={[Function: handleKeyDown]} type="input" defaultValue="" />
          <Spinner classes="hyp-u-absolute-centered">
            <Icon name={[symbol]} containerClasses="Hyp-Spinner--medium hyp-u-absolute-centered">
              <span className="Hyp-Spinner--medium hyp-u-absolute-centered" dangerouslySetInnerHTML={{...}} />
            </Icon>
          </Spinner>
        </span>
```

With local `Spinner` component + tests in the bundle (test passes):

```
        <span className="SubmitGradeForm__grade-wrapper">
          <input className="SubmitGradeForm__grade" disabled={false} id="SubmitGradeForm__grade:1" onInput={[Function: handleKeyDown]} type="input" defaultValue="" />
          <Spinner$1 classes="hyp-u-absolute-centered">
            <Icon name={[symbol]} containerClasses="Hyp-Spinner--medium hyp-u-absolute-centered">
              <span className="Hyp-Spinner--medium hyp-u-absolute-centered" dangerouslySetInnerHTML={{...}} />
            </Icon>
          </Spinner$1>
        </span>
```

Note that in the second case _the spinner is still there_, but it has been renamed so the `wrapper.find(...)` check fails.

This raises the question of why the tests passed before, when using the local `Spinner` implementation. Looking at e9cc7eca06f2c53b3b4b320027f00ff66f350e83, this is because the `isFetchingGrade` function did call `wrapper.update()` prior to `wrapper.find()`.

So to recap:

1. The initial replacement of Spinner in e9cc7eca06f2c53b3b4b320027f00ff66f350e83 should have caused the tests to fail, but didn't because of the renaming of the imported Spinner that Rollup did, due to having multiple components of the same name in the bundle.
2. When the local Spinner implementation was removed, Rollup stopped renaming the imported one, as there was no longer a conflict. This revealed the issue from (1).

**Comment by robertknight** [comment]
Will come back to the other issue (SVG types) on Monday.

**Review by robertknight — APPROVED** [review]
One minor naming suggestion. Otherwise LGTM.

File: lms/static/scripts/frontend_apps/components/SubmitGradeForm.js [hunk 1]
```diff
@@ -190,13 +189,7 @@ export default function SubmitGradeForm({ student }) {
             defaultValue={grade}
             key={student ? student.LISResultSourcedId : null}
           />
-          <Spinner
-            className={classNames('SubmitGradeForm__fetch-spinner', {
-              'is-active': gradeLoading,
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/SubmitGradeForm.js:195]: I didn't feel like the transition added much, and wasn't consistent with our other patterns. The spinner is indeed a little chunky for the space, but is rendering at the same size as the previous spinner. Yes, this component is due for another review pass for the other styling oddities...hope to get to that soon!
